modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,345 @@
+    const [ promise1, promise2, promise3 ] = [
+      editor.userLookup.fetchUsers([ userId ])[0],
+      editor.userLookup.fetchUsers([ userId ])[0],
+      editor.userLookup.fetchUsers([ userId ])[0]
+    ];
+
+    await Promise.all([
+      expect(promise1).to.eventually.deep.equal(await promise2),
+      expect(promise2).to.eventually.deep.equal(await promise3),
+    ]);
+  });
+
+  it('TINY-11974: Should handle invalid user IDs gracefully', async () => {
+    const editor = hook.editor();
+    const invalidIds = [ '', ' ', undefined, null, false, 123 ] as any[];
+
+    // Test each invalid ID asynchronously
+    await Promise.all(Arr.map(invalidIds, async (invalidId) => {
+      const [ promise ] = editor.userLookup.fetchUsers([ invalidId ]);
+      await expect(promise).to.be.rejectedWith(
+        `User ${invalidId} not found`,
+        `Should reject for invalid ID: ${invalidId}`
+      );
+    }));
+  });
+
+  it('TINY-11974: Should maintain separate caches for different user IDs', async () => {
+    const editor = hook.editor();
+    const userId1 = 'test-user-1';
+    const userId2 = 'test-user-2';
+
+    const [ promise1 ] = editor.userLookup.fetchUsers([ userId1 ]);
+    const [ promise2 ] = editor.userLookup.fetchUsers([ userId2 ]);
+
+    await Promise.all([
+      expect(promise1).to.eventually.have.property('id').that.equals(userId1),
+      expect(promise2).to.eventually.have.property('id').that.equals(userId2),
+      expect(promise1).to.eventually.not.equal(promise2)
+    ]);
+  });
+
+  it('TINY-11974: Should handle large batches of users efficiently', async () => {
+    const editor = hook.editor();
+    const userIds = Arr.range(100, (i) => `test-user-${i}`);
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+    const users = await Promise.all(promises);
+
+    await Promise.all(Arr.map(users, async (user, index) =>
+      expect(Promise.resolve(user)).to.eventually.have.property('id').that.equals(`test-user-${index}`)
+    ));
+  });
+
+  it('TINY-11974: Should handle custom user properties', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-custom';
+    const originalFetchUsers = editor.options.get('fetch_users');
+    const customUser = {
+      id: userId,
+      name: 'Test User',
+      avatar: 'test-avatar.png',
+      description: 'Test Description',
+      custom: {
+        role: 'admin',
+        department: 'IT'
+      }
+    };
+
+    try {
+      // Override fetch_users for this test
+      editor.options.set('fetch_users', () => Promise.resolve([ customUser ]));
+
+      const [ userPromise ] = editor.userLookup.fetchUsers([ userId ]);
+
+      await expect(userPromise).to.eventually.deep.equal(customUser);
+      await expect(userPromise).to.eventually.have.deep.property('custom', customUser.custom);
+    } finally {
+      // Restore the original fetch_users function we overrode
+      editor.options.set('fetch_users', originalFetchUsers);
+    }
```

Comment:
This would be better implemented as a `context` block, and inside put this code in an `afterEach`